### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by bur1to**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 const mongoose = require('mongoose');
 const express = require('express');
 const userRouter = require('./routers/users');
+const commnetRouter = require('./routers/comments');
```
Comment (edwardntn) [inline app.js:4]: Please fix a typo "commnetRouter", should be "commentRouter"

File: controllers/authorization.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const User = require('../models/user');
+const crypto = require('crypto');
+const salt = '5d9afb187818d05848770d0c804a7f1f';
```
Comment (edwardntn) [inline controllers/authorization.js:3]: Please move salt to the project settings.

Comment (bur1to) [reply controllers/authorization.js:3]: I don't understand where I need to move salt
Please, can you explain where I need to move salt?

File: controllers/authorization.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const authorization = (async (req, res, next) => {
+  try {
+    const { body } = req;
+
+    const user = await User.findOne({ email: body.email});
+
+    if (!user) {
+      throw new Error('User with this email does not exist');
```
Comment (edwardntn) [inline controllers/authorization.js:12]: Please change the error message to "Incorrect email or password"

File: controllers/authorization.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+   body.password = crypto.pbkdf2Sync(body.password, salt, 1000, 64, 'sha512').toString('hex');
+
+    if (user.password !== body.password) {
+      throw new Error('Invalid password');
```
Comment (edwardntn) [inline controllers/authorization.js:18]: The same as above

File: controllers/users.js [hunk 5]
```diff
@@ -18,7 +19,11 @@ const getUser = (async (req, res, next) => {
 
     const data = await User.findOne({ _id: id });
 
-    res.json(data);
+    if (!data) {
+      throw new Error('User does not exit!');
```
Comment (edwardntn) [inline controllers/users.js:23]: Please change the error message to "User not found"

File: controllers/comments.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  } catch (err) {
+    next(err);
+  }
+});
+
+const createComment = (async (req, res, next) => {
+  try {
+    const { body } = req;
+
+    if (!body) {
```
Comment (edwardntn) [inline controllers/comments.js:35]: You don't need this validation. Just check if userId exists.

**Review by edwardntn — APPROVED** [review]

**Pull request merged**